components/tidb_query_codegen/src/rpn_function.rs

```diff
@@ -1581,7 +1581,13 @@ impl NormalRpnFn {
                 return Ok(#vec_type::cast_chunk_into_vector_value(result));
             }
 
-            for (row_index, val) in BitAndIterator::new(vecs.as_slice(), output_rows).enumerate() {
+            let merged_bitmap = vecs
+                .iter()
+                .fold(BitVec::from_elem(output_rows, true), |mut acc, x| {
+                    acc.and(x);
+                    acc
```

Comment:
Can we use something like `BitAndIterator` before? It seems that using `iter` and `fold` will lead to an extra BitMap allocation upon calling vectorized functions.

Reply:
Sure, I can get `BitAndIterator `. My plan was to go with a simple approach on my first pass and look into the performance impact before merging.

Folding with `BitVec::and` should be more efficient than doing similar work element-by-element because it is done on whole blocks. The allocation of the initial value for the fold could be eliminated in case of non-empty `vecs` - the fold can use tail of `vecs` and fold in the tail. Do you think this approach is worth pursuing?

Reply:
My previous approach of BitAndIterator is to bit-and a word at a time, instead of doing it element by element. I guess bit_vec may have similar implementation for such iterator.